tests/model_test.js

```diff
@@ -0,0 +1,506 @@
+'use strict';
+
+require('./helper');
+
+const { expect } = require('chai'),
+  { Sequelize, DataTypes } = require('../source'),
+  sinon = require('sinon'),
+  Op = Sequelize.Op;
+
+const Support = {
+  dropTestSchemas: async function(sequelize) {
+    const queryInterface = sequelize.getQueryInterface();
+    if (!queryInterface.queryGenerator._dialect.supports.schemas) {
+      return this.sequelize.drop({});
+    }
+
+    const schemas = await sequelize.showAllSchemas();
+    const schemasPromise = [];
+    schemas.forEach(schema => {
+      const schemaName = schema.name ? schema.name : schema;
+      if (schemaName !== sequelize.config.database) {
+        schemasPromise.push(sequelize.dropSchema(schemaName));
+      }
+    });
+
+    await Promise.all(schemasPromise.map(p => p.catch(e => e)));
+  }
+}
+
+describe('Model', () => {
+  before(function() {
+    this.clock = sinon.useFakeTimers();
+  });
+
+  after(function() {
+    this.clock.restore();
+  });
+
+  beforeEach(async function() {
+    this.User = this.sequelize.define('User', {
+      username: DataTypes.STRING,
+      secretValue: DataTypes.STRING,
+      data: DataTypes.STRING,
+      intVal: DataTypes.INTEGER,
+      theDate: DataTypes.DATE,
+      aBool: DataTypes.BOOLEAN
+    });
+
+    await this.User.sync({ force: true });
+  });
+
+  describe('constructor', () => {
+    // Edit Reason: CRDB creates its primary index named 'primary'. 
```

Comment:
Sorry I'm confused about this comment. Why is the naming of the primary key relevant here?

Reply:
This test generates two indexes: [UserWithUniqueFieldAliases_user_name_key, primary]
PG generates: [UserWithUniqueFieldAliases_pkey, UserWithUniqueFieldAliases_user_name_key].
The index naming is bad for the test expectation. No problem beyond that.

Reply:
Ah I see the new comment makes sense to me.